app/models/internal_comment.rb

```diff
@@ -6,10 +6,32 @@ class InternalComment < Comment
   # Generate notifications for InternalComment
 
   # If a reviewer/organizer leaves a comment,
-  #  only the other reviewers of proposal get an in-app notification.
+  # only the other reviewers of proposal get an in-app notification.
+  # if the comment contains an @mention, the mentioned party gets a special notification and an email, regardless of whether they are a reviewer.
   def notify
-    reviewers = proposal.reviewers.reject{|r| r.id == user_id }
-    Notification.create_for(reviewers, proposal: proposal, message: "Internal comment on #{proposal.title}")
+    notify_unmentioned_reviewers
+    notify_mentioned_event_staff
+  end
+
+  def notify_unmentioned_reviewers
+    unmentioned_reviewers = proposal.reviewers.reject{|r| r.id == user_id || mentions?(r)}
```

Comment:
I'm wondering what the database interaction looks like for this method and mentions?  My concern is that it's making too many trips to the db.

Reply:
It might be worth having a method that extracts all the short_names from a comment using a String#scan with a regex and then having one query that finds all the mentioned users using a `where(short_name: short_names)`. I think that might cut down the queries and ruby iterating.
It may be that the old code did not set the best example by using a reject when a where.not would have worked. 
Is that what you had in mind @mghaught?

Reply:
@jonsgreen @mghaught I have removed all the database calls in the method that checks if a staff was mentioned, let me know if you think that's enough.